autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/UpdateFailedHandler.java

```diff
@@ -43,57 +41,53 @@
     @Inject
     private CloudbreakCommunicator cloudbreakCommunicator;
 
-    @Inject
-    private FailedNodeRepository failedNodeRepository;
-
     private final Map<Long, Integer> updateFailures = new ConcurrentHashMap<>();
 
     @Override
     public void onApplicationEvent(UpdateFailedEvent event) {
         long autoscaleClusterId = event.getClusterId();
-        LOGGER.debug("Cluster {} failed", autoscaleClusterId);
         Cluster cluster = clusterService.findById(autoscaleClusterId);
         if (cluster == null) {
             return;
         }
         MDCBuilder.buildMdcContext(cluster);
+        LOGGER.debug("Analysing Cluster Status '{}' ", cluster.getStackCrn());
         StackV4Response stackResponse = getStackById(cluster.getStackCrn());
-        if (stackResponse == null) {
-            LOGGER.debug("Suspending cluster {}", autoscaleClusterId);
+        String stackStatus = getStackStatus(stackResponse);
+        if (stackResponse == null || stackStatus.startsWith(STOPPED_STATUSES_PREFIX)) {
```

Comment:
can you create a list with the suspended statuses and check whether contains or not?

Reply:
Agree, replaced functionality to use inbuilt Status::isAvailable and also abstracted CBCluster state sync functionality out of UpdateFailedHandler since there was overlap functionality here in tracking failures and syncing cluster status. Also split the unit-tests to separate cluster sync and update failure tracking.